python/lsst/ts/MTAlignment/mock/mock_t2sa_point_group.py

```diff
@@ -0,0 +1,134 @@
+    the cartesian coordinates, rotation and radius of the body, plus definition
+    of the location of the measuring points. With this information it is
+    possible to compute the location of each individual point or the entire
+    body in the cartesian coordinate system, plus the respective rotations of
+    the body with respect to the cartesian axis.
+
+    Parameters
+    ----------
+    origin : `CartesianCoordinate`
+        The origin of the body.
+    rotation : `BodyRotation`
+        Rotation of the body with respect to the xyz axis.
+    radius : float
+        The radius of the body.
+
+    Attributes
+    ----------
+    origin : `CartesianCoordinate`
+        Position of the body origin in the cartesian coordinate system.
+    rotation : `BodyRotation`
+        Rotation of the body with respect to the xyz axis.
+    radius : `float`
+        Radius of the body (in meter).
+    """
+
+    def __init__(
+        self,
+        origin: CartesianCoordinate,
+        rotation: BodyRotation,
+        radius: float,
+    ) -> None:
+        self.origin = origin
+        self.rotation = rotation
+        self.radius = radius
+        self._point_angles = np.radians(np.array([0.0, 120.0, 240.0]))
+
+    def get_points_position(
+        self,
+    ) -> typing.List[CartesianCoordinate]:
+        """Calculate the cartesian coordinate position of each individual
+        measuring point that belongs to this point group.
+
+        Returns
+        -------
+        `list` of `CartesianCoordinate`
+            The cartesian coordinates of the targets.
+        """
+
+        origin = self.origin.as_array()
+
+        rotation = Rotation.from_rotvec(self.rotation.as_array()).as_matrix()
+
+        return [
+            np.dot(
+                origin + self.radius * np.array([np.sin(angle), np.cos(angle), 0.0]),
+                rotation,
+            )
+            for angle in self._point_angles
+        ]
+
+    def get_point_position(
+        self,
+        point: int,
+    ) -> CartesianCoordinate:
+        """Calculate a single point position.
+
+        Parameters
+        ----------
+        point : `int`
+            Index of the point to get position (0 based index).
+
+        Returns
+        -------
+        `CartesianCoordinate`
+            The cartesian coordinates of the targets.
+        """
+
+        origin = self.origin.as_array()
+
+        rotation = Rotation.from_rotvec(self.rotation.as_array()).as_matrix()
+
+        angle = self._point_angles[point]
+
+        return CartesianCoordinate(
+            *np.dot(
+                origin + self.radius * np.array([np.sin(angle), np.cos(angle), 0.0]),
+                rotation,
+            )
+        )
+
+    def get_number_of_points(self) -> int:
+        """Get the number of points in this point group.
+
+        Returns
+        -------
+        `int`
+            Number of points.
+        """
+        return len(self._point_angles)
```

Comment:
I strongly suggest adding a method that returns a new MockT2SAPointGroup after applying a translation and rotation. That should simplify the mock T2SA code.

Unless I'm missing something, I think it'll be trivial.